Sources/Apollo/RequestCreator.swift

```diff
@@ -124,17 +124,23 @@ extension RequestCreator {
     let operationData = try serializationFormat.serialize(value: fields)
     formData.appendPart(data: operationData, name: "operations")
 
-    var map = [String: [String]]()
-    if files.count == 1 {
-      let firstFile = files.first!
-      map["0"] = ["variables.\(firstFile.fieldName)"]
-    } else {
-      for (index, file) in files.enumerated() {
-        map["\(index)"] = ["variables.\(file.fieldName).\(index)"]
+    var map = [String: [String: String]]()
+    files.enumerated().forEach { (index, file) in
+      var field = map[file.fieldName, default: [:]]
```

Comment:
One thing that keeps throwing me off when I read this is `field.count`, partly because `field` is singular so I don't expect it to have a `count`. Would recommend renaming it something like `fieldsForFieldName` for clarity